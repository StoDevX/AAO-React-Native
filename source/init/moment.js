// @flow
import moment from 'moment-timezone'

// These next values made possible by
// http://stackoverflow.com/questions/42513276

// Tell Moment to round up, so that Moment will always report the
// time until something else in a way that matches the clock time.
// That is, at 5:22pm, Moment will report "8 minutes until 5:30pm"
// until the clock ticks over to 5:23pm.
// If we do Math.floor, we go the other direction: it will report
// "7 minutes until 5:30pm" as soon as 5:22pm starts.
moment.relativeTimeRounding(Math.ceil)

// Tell Moment that a minute is 60 seconds, for reporting purposes.
// It defaults to 45 seconds.
moment.relativeTimeThreshold('s', 60)
