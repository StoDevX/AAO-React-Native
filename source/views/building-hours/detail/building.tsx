import * as React from 'react'
import {ScrollView, StyleSheet, Image} from 'react-native'
import {images as buildingImages} from '../../../../images/spaces'
import type {BuildingType} from '../types'
import type {Moment} from 'moment-timezone'
import * as c from '@frogpond/colors'
import {getShortBuildingStatus} from '../lib'

import {SolidBadge as Badge} from '@frogpond/badge'
import {Header} from './header'
import {ScheduleTable} from './schedule-table'
import {ListFooter} from '@frogpond/lists'
import {LinkTable} from './link-table'

const styles = StyleSheet.create({
	container: {
		alignItems: 'stretch',
		backgroundColor: c.sectionBgColor,
	},
	image: {
		width: undefined,
		height: 100,
	},
})

type Props = {
	info: BuildingType
	now: Moment
	onProblemReport: () => any
}

const BG_COLORS: Record<string, string> = {
	Open: c.moneyGreen,
	Closed: c.salmon,
}

export class BuildingDetail extends React.Component<Props> {
	shouldComponentUpdate(nextProps: Props) {
		return (
			!this.props.now.isSame(nextProps.now, 'minute') ||
			this.props.info !== nextProps.info ||
			this.props.onProblemReport !== nextProps.onProblemReport
		)
	}

	render() {
		let {info, now, onProblemReport} = this.props

		let headerImage =
			info.image && buildingImages.has(info.image)
				? buildingImages.get(info.image)
				: null

		let openStatus = getShortBuildingStatus(info, now)
		let schedules = info.schedule || []
		let links = info.links || []

		return (
			<ScrollView contentContainerStyle={styles.container}>
				{headerImage ? (
					<Image
						accessibilityIgnoresInvertColors={true}
						resizeMode="cover"
						source={headerImage}
						style={styles.image}
					/>
				) : null}

				<Header building={info} />
				<Badge accentColor={BG_COLORS[openStatus]} status={openStatus} />
				<ScheduleTable
					now={now}
					onProblemReport={onProblemReport}
					schedules={schedules}
				/>

				{links.length ? <LinkTable links={links} /> : null}

				<ListFooter
					title={
						'Building hours subject to change without notice\n\nData collected by the humans of All About Olaf'
					}
				/>
			</ScrollView>
		)
	}
}
