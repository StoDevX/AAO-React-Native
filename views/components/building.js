/**
 * All About Olaf
 * Building Hours list element
 */

import React from 'react'
import {
  StyleSheet,
  View,
  Text,
  Image,
} from 'react-native'

import NavigatorScreen from './navigator-screen'
import * as c from './colors'

let styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    margin: 5,
  },
  buildingOpen: {
    height: 100,
    borderColor: c.pastelGreen,
    borderWidth: 5,
    flex: 1,
  },
  buildingClosed: {
    height: 100,
    borderColor: c.strawberry,
    borderWidth: 5,
    flex: 1,
  },
  buildingAlmostClosed: {
    height: 100,
    borderColor: c.mustard,
    borderWidth: 5,
    flex: 1,
  },
  name: {
    color: c.white,
    fontSize: 30,
    textAlign: 'center',
  },
})

// PROPS: imageSrc, open, name
export default class BuildingView extends React.Component {
  render() {
    switch (this.props.open) {
      case 'open':
        return (
          <View style={styles.container}>
            <Image source={{uri: this.props.imageSource}} style={styles.buildingOpen}>
              <Text style={styles.name}>{this.props.name}</Text>
              </Image>
          </View>
        )
        break
      case 'almostClosed':
        return (
          <View style={styles.container}>
            <Image source={{uri: this.props.imageSource}} style={styles.buildingAlmostClosed}>
              <Text style={styles.name}>{this.props.name}</Text>
              </Image>
          </View>
        )
        break
      case 'closed':
        return (
          <View style={styles.container}>
            <Image source={{uri: this.props.imageSource}} style={styles.buildingClosed}>
              <Text style={styles.name}>{this.props.name}</Text>
              </Image>
          </View>
        )
        break
      default:
        return (
          <View></View>
        )
        break
    }
  }
}
