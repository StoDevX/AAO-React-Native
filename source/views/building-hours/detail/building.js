/**
 * @flow
 *
 * <Building /> controls the structure of the detail view.
 */

import React from 'react'
import {ScrollView, StyleSheet, Platform, Image} from 'react-native'
import {buildingImages} from '../../../../images/building-images'
import type {BuildingType} from '../types'
import moment from 'moment-timezone'
import * as c from '../../components/colors'
import {getShortBuildingStatus} from '../lib'

import {Badge} from './badge'
import {Header} from './header'
import {ScheduleTable} from './schedule-table'
import {ListFooter} from '../../components/list'

const styles = StyleSheet.create({
  container: {
    alignItems: 'stretch',
    ...Platform.select({
      android: {
        backgroundColor: c.androidLightBackground,
      },
      ios: {
        backgroundColor: c.iosLightBackground,
      },
    }),
  },
  image: {
    width: null,
    height: 100,
  },
})

type Props = {
  info: BuildingType,
  now: moment,
  onProblemReport: () => any,
}

export class BuildingDetail extends React.Component<void, Props, void> {
  shouldComponentUpdate(nextProps: Props) {
    return (
      !this.props.now.isSame(nextProps.now, 'minute') ||
      this.props.info !== nextProps.info ||
      this.props.onProblemReport !== nextProps.onProblemReport
    )
  }

  render() {
    const {info, now, onProblemReport} = this.props

    const headerImage = info.image && buildingImages.hasOwnProperty(info.image)
      ? buildingImages[info.image]
      : null
    const openStatus = getShortBuildingStatus(info, now)
    const schedules = info.schedule || []

    return (
      <ScrollView contentContainerStyle={styles.container}>
        {headerImage
          ? <Image
              source={headerImage}
              resizeMode="cover"
              style={styles.image}
            />
          : null}

        <Header building={info} />
        <Badge status={openStatus} />
        <ScheduleTable
          schedules={schedules}
          now={now}
          onProblemReport={onProblemReport}
        />

        <ListFooter
          title={
            'Building hours subject to change without notice\n\nData collected by the humans of All About Olaf'
          }
        />
      </ScrollView>
    )
  }
}
