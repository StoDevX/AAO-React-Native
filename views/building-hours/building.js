// @flow
/**
 * All About Olaf
 * Building Hours list element
 */

import React from 'react'
import {
  StyleSheet,
  View,
  Text,
  Image,
} from 'react-native'

import * as c from '../components/colors'
import type {BuildingInfoType} from './types'
import {isBuildingOpen} from './is-building-open'
import {formatBuildingHours} from './format-building-hours'
import CollapsibleBlock from '../components/collapsibleBlock'
import {allBuildingHours} from './all-building-hours.js'
import Dimensions from 'Dimensions'


var openStatus;
var borderColor;

let borderColors = {
  'Open': c.pastelGreen,
  'Almost Closed': c.mustard,
  'Closed': c.strawberry,
}

type PropsType = {
  info: BuildingInfoType,
  image: number,
  name: string,
};



let style = function(){
    return {
      container: {
        flex: 1,
        justifyContent: 'center',
        borderLeftWidth: 5,
        borderRightWidth: 5,
        marginTop: 10,
        height: 100,
        borderColor: getBorderColor(),
      },
      name: {
        color: c.white,
        fontSize: 24,
        textAlign: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.65)',
      },
      inner: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-between',
      },
      status: {
        color: c.white,
        fontSize: 14,
        paddingTop: 2,
        paddingBottom: 2,
        textAlign: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.45)',
      },
      hoursText:{
        textAlign: 'center',
        fontSize: 13,
        justifyContent: 'space-between'
      },
      contents:{
        paddingBottom: 10,
        paddingTop: 5,
        width: getContainerWidth(),
      },
    }
  }

  let getBorderColor = function(){
    borderColor = borderColors[openStatus];
    return borderColor;
  }

  let getContainerWidth = function(){
    return Dimensions.get('window').width;
  }

export default function BuildingView({info, image, name}: PropsType) {
  openStatus = isBuildingOpen(info)
  
  let hours = formatBuildingHours(info)
  let allHours = allBuildingHours(info, style());



  return (
    <CollapsibleBlock style={style()} borderColor={borderColor}>
    <View style={[style().container]}>
      <Image
        source={image}
        style={{width: undefined, height: 100}}
        resizeMode='cover'>
        <View style={style().inner}>
          <Text style={style().name}>{name}</Text>
          <Text style={style().status}>{hours}</Text>
        </View>
      </Image>
    </View>

    <View style={style().contents}>
      <Text style={style().hoursText}>Daily Hours</Text>
      {allHours}
    </View>
    </CollapsibleBlock>
  )
}

BuildingView.propTypes = {
  image: React.PropTypes.number.isRequired,
  info: React.PropTypes.object.isRequired,
  name: React.PropTypes.string.isRequired,
}
