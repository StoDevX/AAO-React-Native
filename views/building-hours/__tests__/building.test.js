// @flow
import React from 'react'
import renderer from 'react/lib/ReactTestRenderer'
import moment from 'moment-timezone'
const m = time => moment.tz(time, 'ddd H:mm', false, 'America/Winnipeg')

import {BuildingView} from '../building'

it('renders a building', () => {
  const now = m('Fri 11:00')
  const info = {
    image: 'building',
    name: 'Building',
    times: {
      hours: {
        Fri: ['10:00', '16:00'],
      },
    },
  }

  const component = renderer.create(
    <BuildingView
      image={0}
      name={info.name}
      info={info}
      now={now}
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
