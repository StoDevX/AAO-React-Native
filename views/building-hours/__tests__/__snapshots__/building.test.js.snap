exports[`test renders a building 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}>
  <View
    style={
      Array [
        Object {
          "borderLeftWidth": 10,
          "borderRightWidth": 10,
          "flex": 1,
          "height": 100,
          "justifyContent": "center",
          "marginTop": 10,
        },
        Object {
          "borderColor": "#CEFFCE",
        },
      ]
    }>
    <Image
      resizeMode="cover"
      source={0}
      style={
        Object {
          "height": 100,
          "width": undefined,
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.65)",
              "color": "#FFFFFF",
              "fontSize": 24,
              "textAlign": "center",
            }
          }>
          Building
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.45)",
              "color": "#FFFFFF",
              "fontSize": 14,
              "paddingBottom": 2,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }>
          10:00am / 4:00pm â€“ Open
        </Text>
      </View>
    </Image>
  </View>
</TouchableOpacity>
`;
